05.wc/wc.rb

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  stdin = $stdin
+  options, filenames = parse_options(ARGV)
+  if filenames.any?
+    files = convert_to_count_contents(read_files(filenames))
+    length = find_max_length(files)
+    show_count_contents(files, options, length)
+  elsif !stdin.eof?
+    stdin_files = []
+    stdin_files << {
+      name: nil,
+      content: stdin.readlines
+    }
+    inputs = convert_to_count_contents(stdin_files)
+    length = 0
+    show_count_contents(inputs, options, length)
+  end
```

Comment:
最終的に `show_count_contents` を呼出すところはかわらないので、変わるところだけ `if` の中で処理できるとよいですね。例えば、

```suggestion
  files, length =
    if filenames.any?
      [convert_to_count_contents(read_files(filenames)), find_max_length(files)]
    else # ファイル名指定がない場合は標準入力からの入力がある、としてもあまり問題はなさそうに思います
      stdin_files =
        [
          {
            name: nil,
            content: stdin.readlines
          }
        ]
      [convert_to_count_contents(stdin_files), 0]
    end
  show_count_contents(files, options, length)
```

などとすればよいかなと思います。